### PR TITLE
Add AreaLight screenshot test level

### PR DESCRIPTION
**Pull request by rgba16f**
Add camera switching ability to script automation.
Add ability to have multiple screenshot tests per level.

Tested via running the RunAtomScreenshotTests.bat script.

### CODE REVIEW

File: Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp [hunk 1]
```diff
@@ -24,12 +26,16 @@
...
 
 #include <AzFramework/Components/ConsoleBus.h>
+#include <AzFramework/Components/CameraBus.h>
 #include <AzFramework/IO/LocalFileIO.h>
 #include <AzFramework/Windowing/NativeWindow.h>
 
 #include <Atom/RPI.Public/Pass/AttachmentReadback.h>
 #include <Atom/RPI.Public/RPISystemInterface.h>
 
+#pragma optimize("", off)
```
Comment (moudgils) [inline Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp:36]: did you mean to keep this in?

File: Gems/ScriptAutomation/Code/Source/ScriptAutomationSystemComponent.cpp [hunk 2]
```diff
@@ -177,12 +177,13 @@ namespace AZ::ScriptAutomation
 
     void ScriptAutomationSystemComponent::SetIdleFrames(int numFrames)
     {
-        AZ_Assert(m_scriptIdleSeconds == 0, "m_scriptIdleFrames is being stomped");
+        AZ_Assert(m_scriptIdleFrames <= 0, "m_scriptIdleFrames is being stomped");
```
Comment (moudgils) [inline Gems/ScriptAutomation/Code/Source/ScriptAutomationSystemComponent.cpp:180]: An interesting assert description about m_scriptIdleFrames being stomped. Could it just not be bad code somewhere?

File: Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp [hunk 3]
```diff
@@ -782,3 +828,4 @@ namespace AZ::ScriptAutomation
         behaviorContext->Method("CaptureBenchmarkMetadata", &Bindings::CaptureBenchmarkMetadata);
     }
 } // namespace AZ::ScriptAutomation
+#pragma optimize("", on)
```
Comment (moudgils) [inline Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp:831]: Is this left on purpose?

File: Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp (hunk 1 above)
Comment (rgba16f) [reply Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp]: no, removed now

File: Gems/ScriptAutomation/Code/Source/ScriptAutomationSystemComponent.cpp (hunk 2 above)
Comment (rgba16f) [reply Gems/ScriptAutomation/Code/Source/ScriptAutomationSystemComponent.cpp:180]: It's not currently being stomped, this is just to catch that happening in the future.

File: Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp (hunk 3 above)
Comment (rgba16f) [reply Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp]: no, removed.

File: Gems/ScriptAutomation/Code/Source/Platform/Windows/Utils_Windows.cpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+#include <AzCore/std/string/string.h>
+#include <AzCore/PlatformIncl.h>
+
+namespace AZ::Platform
+{
+    bool LaunchProgram(const AZStd::string& progPath, const AZStd::string& arguments)
+    {
+            bool result = false;
```
Comment (moudgils) [inline Gems/ScriptAutomation/Code/Source/Platform/Windows/Utils_Windows.cpp:16]: Tabbing issues

**Review by moudgils — APPROVED** [review]

File: Gems/ScriptAutomation/Code/Source/Platform/Windows/Utils_Windows.cpp (hunk 4 above)
Comment (rgba16f) [reply Gems/ScriptAutomation/Code/Source/Platform/Windows/Utils_Windows.cpp]: fixed.

File: Gems/ScriptAutomation/Code/Source/Platform/Windows/Utils_Windows.cpp [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                NULL,           // Use parent's starting directory 
+                &si,            // Pointer to STARTUPINFO structure
+                &pi             // Pointer to PROCESS_INFORMATION structure (removed extra parentheses)
+            );
+            // Close process and thread handles. 
+            CloseHandle(pi.hProcess);
+            CloseHandle(pi.hThread);
+
+            return result;
+
```
[the remark this answers is not in the document]
Comment (rgba16f) [reply Gems/ScriptAutomation/Code/Source/Platform/Windows/Utils_Windows.cpp]: removed.

File: Gems/ScriptAutomation/Code/Source/Platform/Linux/Utils_Linux.cpp [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            AZ_Error(
+                "LaunchProgram", false,
+                "LaunchProgram: Unable to launch executable %s : errno = %s."
+                "line tools.",
+                localProgPathCopy.c_str(), strerror(errno));
+
+            _exit(errno);
+        }
+
+
```
[the remark this answers is not in the document]
Comment (rgba16f) [reply Gems/ScriptAutomation/Code/Source/Platform/Linux/Utils_Linux.cpp]: removed extra line.

**Review by gadams3 — APPROVED** [review]

**Pull request merged**